### PR TITLE
Team invite: adds a warning on shared email detection

### PR DESCRIPTION
**Pull request by sarthaktexas**
_Closes #5166_

## Describe your changes
Screenshot 2024-02-12 at 4 42 50 PM

(@malted you're _my_ favorite rat)

### CODE REVIEW

**Comment by garyhtou** [comment]
> (@malted you're _my_ favorite rat)

🤔, @malted == 🐀?!

File: app/views/events/_form_invite.erb [hunk 1]
```diff
@@ -3,13 +3,16 @@
...
-  <div class="field">
+  <div class="field" x-data="{ email: '' }">
     <%= form.label :email, class: "mb1" %>
 
     <div class="flex items-center">
-      <%= form.email_field :email, class: "mr2" %>
+      <%= form.email_field :email, class: "mr2", "x-model": "email" %>
       <%= form.submit 'Send', disabled: !organizer_signed_in? %>
     </div>
+    <template x-if="email.includes('team', 'webmaster', 'marketing', 'admin')">
```
Comment (garyhtou) [inline app/views/events/_form_invite.erb]: let's add:
- info
- about
- support
- sales
- hq
- hello

**Review by garyhtou — COMMENTED** [review]
coolio!

**Comment by malted** [comment]
> > (@malted you're _my_ favorite rat)
> 
> 
> 
> 🤔, @malted == 🐀?!

image

File: app/views/organizer_position_invites/_form.html.erb [hunk 2]
```diff
@@ -1,19 +1,23 @@
...
+    <div class="field flex items-end">
+      <div class="mr2 flex-auto" >
+        <%= form.label :email %>
+        <%= form.email_field :email, "x-model": "email" %>
+      </div>
+      <%= form.submit "Send" %>
     </div>
-    <%= form.submit "Send" %>
+    <template x-if="email.includes('team', 'webmaster', 'marketing', 'admin')">
+      <div class="warning mt1">⚠️ We discourage team emails on HCB. You can still use this email, but we recommend individual emails like ben@hackclub.com!</div>
```
Comment (malted) [inline app/views/organizer_position_invites/_form.html.erb]:
```suggestion
      <div class="warning mt1">⚠️ We discourage team emails on HCB. You can still use this email, but we recommend individual emails, for example <em>ben@hackclub.com!</em></div>
```

File: app/views/events/_form_invite.erb (hunk 1 above)
Comment (sarthaktexas) [reply app/views/events/_form_invite.erb]: done.

File: app/views/events/_form_invite.erb [hunk 3]
```diff
@@ -3,13 +3,16 @@
...
+  <div class="field" x-data="{ email: '' }">
     <%= form.label :email, class: "mb1" %>
 
     <div class="flex items-center">
-      <%= form.email_field :email, class: "mr2" %>
+      <%= form.email_field :email, class: "mr2", "x-model": "email" %>
       <%= form.submit 'Send', disabled: !organizer_signed_in? %>
     </div>
+    <template x-if="email.includes('team', 'webmaster', 'marketing', 'admin', 'info', 'about', 'support', 'sales', 'hq', 'hello')">
+      <div class="warning mt1">⚠️ We discourage team emails on HCB. You can still use this email, but we recommend individual emails, for example <em>ben@hackclub.com!</em></div>
```
Comment (sampoder) [inline app/views/events/_form_invite.erb]:
```suggestion
      <div class="warning mt1">⚠️ We discourage team emails on HCB. You can still use this email, but we recommend individual emails, for example <em>ben@hackclub.com</em>!</div>
```

File: app/views/organizer_position_invites/_form.html.erb [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
 
 <%= form_with(model: invite, local: true, url: event_organizer_position_invites_path) do |form| %>
   <%= form_errors invite, "user" %>
-
-  <div class="field flex items-end">
-    <div class="mr2 flex-auto">
-      <%= form.label :email %>
-      <%= form.email_field :email %>
+  <div x-data="{ email: '' }">
+    <div class="field flex items-end">
```
Comment (garyhtou) [inline app/views/organizer_position_invites/_form.html.erb]: Let's get rid of this extra div and move the `x-data="{ email: '' }"` into the inner div

File: app/views/events/_form_invite.erb [hunk 5]
```diff
@@ -22,5 +22,8 @@
 
       <%= form.submit 'Send', disabled: !organizer_signed_in? %>
     </div>
+    <template x-if="email.includes('team', 'webmaster', 'marketing', 'admin', 'info', 'about', 'support', 'sales', 'hq', 'hello')">
```
Comment (garyhtou) [inline app/views/events/_form_invite.erb]: Maybe these should be `team@`, `webmaster@`, etc.

Comment (sarthaktexas) [reply app/views/events/_form_invite.erb]: done!

File: app/views/organizer_position_invites/_form.html.erb [hunk 6]
```diff
@@ -1,19 +1,23 @@
...
-    <%= form.submit "Send" %>
+    <template x-if="email.includes('team@', 'webmaster@', 'marketing@', 'admin@', 'info@', 'about@', 'support@', 'sales@', 'hq@', 'hello@')">
+      <div class="warning mt1">⚠️ We discourage team emails on HCB. We recommend individual emails such as <em>ben@hackclub.com</em>!</div>
+    </template>
+    <% admin_tools("mt1") do %>
+      <div class="field field--checkbox m0">
+        <%= form.check_box :is_signee %>
+        <%= form.label :is_signee, "Signed contract" %>
+      </div>
+    <% end %>
```
Comment (garyhtou) [inline app/views/organizer_position_invites/_form.html.erb]: Let's move this back outside the div

**Review by garyhtou — APPROVED** [review]

**Pull request merged**